### PR TITLE
Appointments page updated v2

### PR DESCRIPTION
**Pull request by anand-ts**
- Updated Appointments Schema
- Updated Appointments Controller
- Updated Appointments Page

### CODE REVIEW

File: server/model/appointment.js [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+// 	createdAt: {
+// 		type: Date,
+// 		default: Date.now,
+// 	},
+// });
+
+// const Appointment = mongoose.model("Appointment", AppointmentSchema);
+
+// export default Appointment;
+
```
Comment (DavidHe137) [inline server/model/appointment.js:27]: Can this be deleted?

File: server/route/AppointmentController.js [hunk 2]
```diff
@@ -1,3 +1,226 @@
...
+// 				available: appointmentCount < 3,
+// 			});
+// 		}
+
+// 		res.json(availability);
+// 	})
+// );
+
+// export default router;
+
```
Comment (DavidHe137) [inline server/route/AppointmentController.js:223]: Can this be deleted?

File: server/route/AppointmentController.js [hunk 3]
```diff
@@ -53,16 +276,22 @@ router.post(
...
 			duration,
 			notes,
 		});
 
 		const savedAppointment = await newAppointment.save();
+
+		// Populate client and stylist details before sending the response
+		await savedAppointment
+			.populate("client", "info.name")
+			.populate("stylist", "info.name");
```
Comment (DavidHe137) [inline server/route/AppointmentController.js:293]: When I test create appointment using Postman, I get the following error:
Screen Shot 2024-11-07 at 12 34 27 PM

**Review by DavidHe137 — CHANGES_REQUESTED** [review]
Backend changes look pretty good, though there seems to be a bug with create appointment. Even though the API has not been fully connected with the backend, were you able to fully test the flow with with curl/Postman/Insomnia?

Also, please delete the commented code as we can use git history to keep track of past code versions.

File: server/route/AccountController.js [hunk 4]
```diff
@@ -1,320 +1,112 @@
...
 
+// Endpoint to upload profile photo
 router.post(
-	"/signIn",
-	[
-		body("username").not().isEmpty().trim().escape(),
-		body("password").not().isEmpty().trim().escape(),
-	],
+	"/photo",
+	upload.single("photo"),
```
Comment (DavidHe137) [inline server/route/AccountController.js:19]: Why was this endpoint deleted?

File: server/model/photo.js [hunk 5]
```diff
@@ -12,4 +12,4 @@ const PhotoSchema = new mongoose.Schema({
 
 const Photo = mongoose.model("Photo", PhotoSchema);
 
-export { Photo, PhotoSchema };
+export default Photo;
```
Comment (DavidHe137) [inline server/model/photo.js:15]: This is breaking imports. I am unable to start the backend with this change.

**Review by DavidHe137 — CHANGES_REQUESTED** [review]
I am unable to run the backend with the original code. After fixing imports, signing in is broken as it seems this PR deletes the endpoint. Were you able to test on your machine?

**Comment by anand-ts** [comment]
Closing this PR since it got too convoluted because I combined Appointments Page and Photo Upload in one PR. Also this breaks Account Creation (my bad) so Appointments will have its own dedicated PR again.

**Pull request closed without merging**